### DISCUSSION TITLE
Frontend para validar a aplicação

### DISCUSSION THREAD

**Discussion by julioflima** [BrasilAPI/BrasilAPI#234] [Ideas]
Sei que por tratar-se de uma API o frontend talvez não seja a BIGPICTURE aqui, mas acredito que alguns forms para o teste da plataforma dentro do site como propõe por exemplo uma outra api que faz captura de dados por CNPJ:
https://www.receitaws.com.br/api

Sejam fundamentais para validar a ideia por quem se propõe a utilizar, sem ter que gerar um enpoint no NPM para validar. (apesar de que isso já é muito bom)

Assim como a implementação da API em várias linguagens com a demonstração em código em cada uma das linguagens, como faz:
https://www.remove.bg/api

E o próprio Google na sua documentação do GCP:
https://cloud.google.com/datastore/docs/concepts/queries

**Comment by wilsonneto-swe** [comment]
@juloko Acredito que utilizar o swagger seria bem legal:
[https://app.swaggerhub.com/apis/ft-phillip-proulx/Tricon-Smart-Action/1.0](https://app.swaggerhub.com/apis/ft-phillip-proulx/Tricon-Smart-Action/1.0)

Já utilizei plugins do express que geravam ele de maneira automática, havia um "/api/docs" onde eram descritos todos os endpoints, entidades, eschemas, respostas e é possível efetuar testes.

Talvez seria uma solução interessante aqui, o que acham?

**Comment by julioflima** [comment]
Sobre isso teve um cara que postou várias soluções, não me pareceu que o swagger era o melhor. Mas de documentação creio que seja esse o caminho.

Mas ainda seria legal uma Single Page Screen, com uns forms para testar e a implementação da API em várias linguagens..

**Comment by wilsonneto-swe** [comment]
Lembra qual julgava ser melhor que o swagger, a vantagem dele é que é o mais utilizado e a maioria das techs tem maneiras e libs que geram o open api automático para você, e como dev eu prefiro consumir ele que qualquer outro, pois já estou acostumado a documentar e a consumir documentação de apis com ele.

Mas se tiver outra ferramenta legal manda ai, e se definirmos uma legal, vamos trabalhar em um PR implementando ela, o que acha?

**Comment by julioflima** [comment]
@wilsonneto-dev 

Acredito que a #14 tenha as pessoas corretas e seja o local correto pra discutir sobre isso. Mas pela tua paixão, vou dar uma olhada nele e tentar implementar em alguma aplicação minha. Vlw!

**Comment by wilsonneto-swe** [comment]
@juloko hehe mas nada, sem paixões hehe se julgar algum legal, bora! Vou ver lá a outra issue, valeu!!

**Comment by Nikolassantos** [comment]
Penso que seria muito interessante utilizarmos React para fazer o Front-End com Consultar CEP e até para ser uma parte gráfica da aplicação para ficar mais dinâmica a consulta.
